entity/volume_group.go

```diff
@@ -2,17 +2,44 @@ package entity
 
 // VolumeGroup represents the MAAS VolumeGroup endpoint.
 type VolumeGroup struct {
-	LogicalVolumes     interface{} `json:"logical_volumes,omitempty"`
-	Devices            interface{} `json:"devices,omitempty"`
-	HumanSize          string      `json:"human_size,omitempty"`
-	UUID               string      `json:"uuid,omitempty"`
-	HumanAvailableSize string      `json:"human_available_size,omitempty"`
-	SystemID           string      `json:"system_id,omitempty"`
-	ResourceURI        string      `json:"resource_uri,omitempty"`
-	HumanUsedSize      string      `json:"human_used_size,omitempty"`
-	Name               string      `json:"name,omitempty"`
-	Size               int64       `json:"size,omitempty"`
-	UsedSize           int64       `json:"used_size,omitempty"`
-	AvailableSize      int64       `json:"available_size,omitempty"`
-	ID                 int         `json:"id,omitempty"`
+	Devices            interface{}     `json:"devices,omitempty"`
+	ResourceURI        string          `json:"resource_uri,omitempty"`
+	HumanSize          string          `json:"human_size,omitempty"`
+	UUID               string          `json:"uuid,omitempty"`
+	HumanAvailableSize string          `json:"human_available_size,omitempty"`
+	SystemID           string          `json:"system_id,omitempty"`
+	HumanUsedSize      string          `json:"human_used_size,omitempty"`
+	Name               string          `json:"name,omitempty"`
+	LogicalVolumes     []LogicalVolume `json:"logical_volumes,omitempty"`
+	Size               int64           `json:"size,omitempty"`
+	UsedSize           int64           `json:"used_size,omitempty"`
+	AvailableSize      int64           `json:"available_size,omitempty"`
+	ID                 int             `json:"id,omitempty"`
+}
+
+// VolumeGroupParams enumerates the parameters for the volume group operation
```

Comment:
You should use a different Params struct for update. Copied from the source code [link-1](https://github.com/canonical/maas/blob/master/src/maasserver/forms/__init__.py#L4213) [link-2](https://github.com/canonical/maas/blob/master/src/maasserver/api/volume_groups.py#L209):

```py
name = forms.CharField(required=False)
uuid = UUID4Field(required=False)
add_block_devices = forms.MultipleChoiceField(required=False)
remove_block_devices = forms.MultipleChoiceField(required=False)
add_partitions = forms.MultipleChoiceField(required=False)
remove_partitions = forms.MultipleChoiceField(required=False)
```
So we need something like:

```go
type VolumeGroupUpdateParams struct {
	Name               string   `url:"name,omitempty"`
	UUID               string   `url:"uuid,omitempty"`
	AddBlockDevices    []string `url:"add_block_devices,omitempty"`
        RemoveBlockDevices []string `url:"remove_block_devices,omitempty"`
	AddPartitions      []string `url:"add_partitions,omitempty"`
        RemovePartitions   []string `url:"remove_partitions,omitempty"`
}
```

Reply:
Fixed in [1440eb0](https://github.com/canonical/gomaasclient/pull/94/commits/1440eb029a45da1bf36b1464eddaa2916e68c2a3).